src/v/resource_mgmt/cpu_profiler.cc

```diff
@@ -90,26 +87,35 @@ ss::future<std::vector<cpu_profiler::shard_samples>> cpu_profiler::results(
                 std::move(shard_result.samples));
               return results;
           });
+        // sort by shard ID so the shard id lines up with the vector index
+        std::ranges::sort(
+          results, [](auto& l, auto& r) { return l.shard < r.shard; });
     }
 
     co_return results;
 }
 
 cpu_profiler::shard_samples cpu_profiler::shard_results(
   std::optional<ss::lowres_clock::time_point> filter_before) const {
-    size_t dropped_samples = 0;
-    absl::node_hash_map<ss::simple_backtrace, size_t> backtraces;
+    size_t dropped_samples = 0, total_samples = 0;
+    chunked_hash_map<ss::simple_backtrace, size_t> backtraces;
```

Comment:
As mentioned on slack I can see chunked_hash_map also suffering from this in general if the type is explicitly absl hashable and the seed is stored in the hasher and not somewhere else.

Reply:
let me follow up on slack

Reply:
I should add I did test this both with cpu_profiler and in a (not submitted) unit test which showed (non-exhaustively) that is stable with chunked map.

Also, the hash here is in seastar so it is not absl-aware, must use only std::hash<>.

To summarize slack: hash codes should be stable within a process, so we are fine to use absl hash with chunked map and expect deterministic order (caveats apply).